From linux-kernel-owner@vger.kernel.org Wed May 23 01:44:17 2007
Return-Path:
	<linux-kernel-owner+tglx=40linutronix.de-S1759353AbXEVXoG@vger.kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.1.7-deb (2006-10-05) on debian
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=none autolearn=unavailable 
	version=3.1.7-deb
Received: from vger.kernel.org (vger.kernel.org [209.132.176.167]) by
	mail.tglx.de (Postfix) with ESMTP id 32C4A65C3E9 for <tglx@linutronix.de>;
	Wed, 23 May 2007 01:44:17 +0200 (CEST)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand id
	S1759353AbXEVXoG (ORCPT <rfc822;tglx@linutronix.de>); Tue, 22 May 2007
	19:44:06 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id
	S1757791AbXEVXn4 (ORCPT <rfc822;linux-kernel-outgoing>); Tue, 22 May 2007
	19:43:56 -0400
Received: from rwcrmhc11.comcast.net ([204.127.192.81]:35206 "EHLO
	rwcrmhc11.comcast.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org with
	ESMTP id S1757669AbXEVXn4 (ORCPT <rfc822;Linux-kernel@vger.kernel.org>);
	Tue, 22 May 2007 19:43:56 -0400
Received: from sx.thebigcorporation.com ([69.181.45.228]) by comcast.net
	(rwcrmhc11) with ESMTP id <20070522233624m1100rg2vge>; Tue, 22 May 2007
	23:36:29 +0000
Received: from sx.thebigcorporation.com (localhost.localdomain [127.0.0.1])
	by sx.thebigcorporation.com (8.13.8/8.13.8) with ESMTP id l4MNaKHv029409;
	Tue, 22 May 2007 16:36:20 -0700
Received: (from sven@localhost) by sx.thebigcorporation.com
	(8.13.8/8.13.8/Submit) id l4MNaJIn029408; Tue, 22 May 2007 16:36:19 -0700
X-Authentication-Warning: sx.thebigcorporation.com: sven set sender to
	sven@thebigcorporation.com using -f
Subject: [PATCH] 2.6.21-rt6
From:	Sven-Thorsten Dietrich <sven@thebigcorporation.com>
To:	LKML <Linux-kernel@vger.kernel.org>
Cc:	Ingo Molnar <mingo@elte.hu>
In-Reply-To: <1179874795.25500.40.camel@sx.thebigcorporation.com>
References: <1179874795.25500.40.camel@sx.thebigcorporation.com>
Content-Type: text/plain
Organization: The Big Corporation
Date:	Tue, 22 May 2007 16:36:19 -0700
Message-Id: <1179876979.25500.54.camel@sx.thebigcorporation.com>
Mime-Version: 1.0
X-Mailer: Evolution 2.8.3 (2.8.3-2.fc6) 
Sender:	linux-kernel-owner@vger.kernel.org
Precedence: bulk
X-Mailing-List:	linux-kernel@vger.kernel.org
X-Filter-To: .Kernel.LKML
X-Evolution-Source: imap://tglx%40linutronix.de@localhost:8993/
Content-Transfer-Encoding: 8bit

On Tue, 2007-05-22 at 15:59 -0700, Sven-Thorsten Dietrich wrote:
> Add 
<correct>
> header and export for rt_write_trylock_irqsave.

Disregard the last patch, flags parameter was missing in the header.

---
 include/linux/spinlock.h |    2 ++
 kernel/rt.c              |    1 +
 2 files changed, 3 insertions(+)

Index: linux-2.6.23.1-rt5/include/linux/spinlock.h
===================================================================
--- linux-2.6.23.1-rt5.orig/include/linux/spinlock.h
+++ linux-2.6.23.1-rt5/include/linux/spinlock.h
@@ -294,6 +294,8 @@ do {									\
 extern void __lockfunc rt_write_lock(rwlock_t *rwlock);
 extern void __lockfunc rt_read_lock(rwlock_t *rwlock);
 extern int __lockfunc rt_write_trylock(rwlock_t *rwlock);
+extern int __lockfunc rt_write_trylock_irqsave(rwlock_t *trylock,
+					       unsigned long *flags);
 extern int __lockfunc rt_read_trylock(rwlock_t *rwlock);
 extern void __lockfunc rt_write_unlock(rwlock_t *rwlock);
 extern void __lockfunc rt_read_unlock(rwlock_t *rwlock);
Index: linux-2.6.23.1-rt5/kernel/rt.c
===================================================================
--- linux-2.6.23.1-rt5.orig/kernel/rt.c
+++ linux-2.6.23.1-rt5/kernel/rt.c
@@ -172,6 +172,7 @@ int __lockfunc rt_write_trylock_irqsave(
 	*flags = 0;
 	return rt_write_trylock(rwlock);
 }
+EXPORT_SYMBOL(rt_write_trylock_irqsave);
 
 int __lockfunc rt_read_trylock(rwlock_t *rwlock)
 {
